src/Components/ArtsyShippingEstimate.tsx

```diff
@@ -0,0 +1,366 @@
+        setContext({ enabled: false, loaded: true, Arta: null })
+      }
+    }
+    initArta()
+  }, [context.loaded, widgetWillLoad])
+
+  return (
+    <ArtsyShippingEstimateContext.Provider value={context}>
+      {props.children}
+    </ArtsyShippingEstimateContext.Provider>
+  )
+}
+
+type State =
+  | {
+      loaded: false
+    }
+  | {
+      loaded: true
+      widget: ArtaEstimate | null
+    }
+
+interface Props {
+  artwork: ArtsyShippingEstimate_artwork$key
+  fallbackText: string
+}
+
+const Loading = () => (
+  <SkeletonText variant="sm">Estimate shipping</SkeletonText>
+)
+
+export const ArtsyShippingEstimate = ({ artwork }: Props) => {
+  const artworkData = useFragment(ARTWORK_FRAGMENT, artwork)
+
+  const estimateInput = estimateRequestBodyForArtwork(
+    artworkData as ShippableArtwork,
+  )
+  console.log("***", "estimateInput", estimateInput)
+  const initialState: State = !!estimateInput
+    ? { loaded: false }
+    : { loaded: true, widget: null }
+
+  const artaContext = useContext(ArtsyShippingEstimateContext)
+  const [state, setState] = useState<State>(initialState)
+
+  const {
+    loaded: isArtaInitialized,
+    enabled: isArtaEnabled,
+    Arta,
+  } = artaContext
+
+  useEffect(() => {
+    console.log("*** loading estimate widget...", {
+      ...state,
+      isArtaInitialized,
+      isArtaEnabled,
+      estimateInput,
+    })
+    if (!isArtaInitialized || state.loaded) {
+      return
+    }
+
+    const loadEstimate = async () => {
+      console.log("***", "ArtsyShippingEstimate params", estimateInput)
+
+      const artsyEstimateWidget =
+        isArtaEnabled && estimateInput && Arta.estimate(estimateInput)
+
+      if (!artsyEstimateWidget) {
+        setState({ loaded: true, widget: null })
+        return
+      }
+
+      try {
+        await artsyEstimateWidget.validate()
+        setState({ loaded: true, widget: artsyEstimateWidget })
+        return
+      } catch (e) {
+        console.error("*** Error validating artsy shipping estimate:", e)
+        setState({ loaded: true, widget: null })
+      }
+    }
+    loadEstimate()
+  }, [
+    isArtaInitialized,
+    isArtaEnabled,
+    Arta,
+    estimateInput,
+    state.loaded,
+    state,
+  ])
+
+  if (!state.loaded) {
+    return <Loading />
+  }
+
+  if (!state.widget) {
+    return <Text>Shipping cost calculated at checkout</Text>
+  }
+
+  const estimateWidget = state.widget
+
+  return estimateWidget.isReady ? (
+    <Link onClick={() => estimateWidget.open()}>Estimate Shipping</Link>
+  ) : (
+    <Loading />
+  )
+}
+
+const ARTWORK_FRAGMENT = graphql`
+  fragment ArtsyShippingEstimate_artwork on Artwork {
+    isFramed
+    category # Raw category required for mapping to Arta subtype
+    shippingOrigin
+    shippingCountry
+    location {
+      country
+      postalCode
+      city
+    }
+    priceCurrency
+    priceListed {
+      major
+    }
+    heightCm
+    widthCm
+    # shippingWeight # may need to be added?
+  }
+`
```

Comment:
Problems with our schema:
- `category` is deprecated but used to map to values required for estimate
- weight not accessible
- location not accessible, so using `shippingOrigin`
- `priceListed` does not include `major` (see https://github.com/artsy/metaphysics/pull/6367)